### PR TITLE
Fix olympia user uid docker for mac

### LINKED ISSUE #2616: Docker for Mac doesn't like the entrypoint bash script
Using the (in beta) Docker for Mac, our script that creates a new `olympia` user fails for the worker image and means we can't boot docker/addons-server.

If I remove this line: https://github.com/mozilla/addons-server/blob/master/docker-compose.yml#L4

It's all fine; everything starts up no problem and I can run `make initialize_docker`, then navigate to `0.0.0.0` and get Add-ons site.

Not sure if that setting messes things up for Linux users who aren't using a VM, would love to know. But it might be a relic of the old Virtualibox way of using Docker which it seems is being phased out.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #2616

### CODE REVIEW

File: scripts/start-docker.sh [hunk 1]
```diff
@@ -11,7 +11,18 @@ uid=$(ls -nd . | awk '{ print $3 }')
 
 # Create an `olympia` user with that ID, and the current directory
 # as its home directory.
-useradd -Md $(pwd) -u $uid olympia
+if [[ $uid -eq 0 ]]; then
+    # Don't try and create a user with uid 0 since it will fail.
+    # Works around issue with docker for mac running containers
+    # as root and not the user. Instead we just create the olympia
+    # user since files will transparently be still owned by the host
```
Comment (tofumatt) [inline scripts/start-docker.sh]: "files will transparently be still owned" is awkwardly phrased, I think "will still be owned" would do the trick?

**Comment by tofumatt** [comment]
r+wc

**Comment by muffinresearch** [comment]
@tofumatt assuming you haven't got anything important in your docker for mac env, perhaps you could try resetting it (which deletes everything as a scorched earth approach) and try the docker setup with this branch. 

Letting an olympia user be created with the next available uid seems to work, and I see all the `*-all.css` files created by the make initialize_docker as being owned by my user *not* the olympia uid.

File: scripts/start-docker.sh (hunk 1 above)
Comment (muffinresearch) [inline scripts/start-docker.sh]: Heh, sure np

**Comment by tofumatt** [comment]
I've been using `docker-machine` lately, so I'll give it a go.

**Comment by muffinresearch** [comment]
Bear in mind the reset will torch everything if you use it 🔥 🔥 🚒

**Comment by muffinresearch** [comment]
Also if you don't use it and you somehow already have an olympia user it won't prove anything since the useradd command will just show an error but the script will continue on.

**Comment by tofumatt** [comment]
Just ran this on Docker for Mac and sadly got this:

```
~/Projects/addons-server[master❗ ]% docker logs addonsserver_web_1
useradd: UID 0 is not unique
su: user olympia does not exist
useradd: UID 0 is not unique
su: user olympia does not exist
```

😢

**Comment by muffinresearch** [comment]
That output looks like you were running against master rather than this branch.

**Comment by tofumatt** [comment]
Indeed, I'm silly.

Okay, so I ran it and it worked great! Could upload an add-on and check it against the validator too, so all the containers are doing their job. 👍 

r+

**Pull request merged**